src/bonjour.c

```diff
@@ -0,0 +1,186 @@
+    const unsigned char *txtRecord,
+    void *context)
+{
+    if (errorCode != kDNSServiceErr_NoError)
+    {
+        fprintf(stderr, "Bonjour Resolve Callback, DNSServiceErrorType %d", errorCode);
+        return;
+    }
+
+    char ip_address[INET_ADDRSTRLEN];
+    resolve_ip_address(hosttarget, port, ip_address);
+
+    browse_data_t *browse_data = (browse_data_t *)context;
+    lxi_info_t *info = (lxi_info_t *)browse_data->info;
+
+    // Pretty print service type
+    const char *service_type = "Unknown";
+    for (lxi_service_t *s = lxi_services; s->broadcast_type != NULL; s++)
+    {
+        if (strcmp(browse_data->regtype, s->broadcast_type) == 0)
+        {
+            service_type = s->service_name;
+            break;
+        }
+    }
+
+    info->service(ip_address, browse_data->servicename, service_type, ntohs(port));
+
+    free(browse_data->regtype);
+    free(browse_data->servicename);
+    free(browse_data);
+}
+
+void browse_callback(
+    DNSServiceRef service,
+    DNSServiceFlags flags,
+    uint32_t interfaceIndex,
+    DNSServiceErrorType errorCode,
+    const char *serviceName,
+    const char *regtype,
+    const char *replyDomain,
+    void *context)
+{
+    if (errorCode != kDNSServiceErr_NoError)
+    {
+        fprintf(stderr, "Bonjour browse Callback, DNSServiceErrorType %d", errorCode);
+        return;
+    }
+
+    char if_name[IF_NAMESIZE];
+    if_indextoname((unsigned int)interfaceIndex, if_name);
+
+    lxi_info_t *info = (lxi_info_t *)context;
+    info->broadcast(serviceName, if_name);
+
+    browse_data_t *browse_data = (browse_data_t *)malloc(sizeof(browse_data_t));
+    if (!browse_data)
+    {
+        printf("Browse data struct memory allocation failed\n");
+        return;
+    }
+
+    browse_data->info = info;
+    browse_data->servicename = strdup(serviceName);
+    browse_data->regtype = strdup(regtype);
+
+    DNSServiceRef resolveService;
+    DNSServiceErrorType error;
+
+    error = DNSServiceResolve(&resolveService, 0, interfaceIndex, serviceName, regtype, replyDomain, resolve_callback, browse_data);
+    if (error != kDNSServiceErr_NoError)
+    {
+        fprintf(stderr, "DNSServiceResolve() failed: %d", error);
+        return;
+    }
+
+    error = DNSServiceProcessResult(resolveService);
+    if (error != kDNSServiceErr_NoError)
+    {
+        fprintf(stderr, "DNSServiceProcessResult() failed: %d", error);
+        return;
+    }
+
+    DNSServiceRefDeallocate(resolveService);
+}
+
+void browse_lxi_services(lxi_info_t *info, int timeout_ms)
+{
+    DNSServiceRef service;
+    DNSServiceErrorType error;
+
+    struct timeval timeout;
+    timeout.tv_sec = timeout_ms / 1000;           // convert milliseconds to seconds
+    timeout.tv_usec = (timeout_ms % 1000) * 1000; // remainder in microseconds
+
+    for (lxi_service_t *s = lxi_services; s->broadcast_type != NULL; s++)
+    {
+        error = DNSServiceBrowse(&service, 0, 0, s->broadcast_type, "local.", browse_callback, info);
```

Comment:
I think we should just use NULL here instead "local." as the documentation states the following:
```
domain

    If non-NULL, specifies the domain on which to browse for services. Most applications will not specify a domain, instead browsing on the default domain(s).
```

This way it falls back to using whatever is hosts default.